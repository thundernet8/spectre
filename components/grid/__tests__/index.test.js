import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { Row, Col } from "..";

describe("Grid", () => {
    it("should render Row", () => {
        const row = shallow(<Row />);

        expect(row.type()).toEqual("div");

        expect(row.hasClass("spt-row")).toBe(true);
    });

    it("should render Col", () => {
        const col = shallow(<Col span={6} />);

        expect(col.type()).toEqual("div");

        expect(col.hasClass("spt-col-6")).toBe(true);
    });

    it("renders correctly", () => {
        const tree = renderer
            .create(
                <Row gutter={10}>
                    <Col span={4}>Left Col</Col>
                    <Col span={20}>Right Col</Col>
                </Row>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
