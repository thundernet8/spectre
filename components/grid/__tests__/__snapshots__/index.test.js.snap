// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders correctly 1`] = `
<div
  className="spt-row spt-row-flex"
  style={
    Object {
      "marginLeft": -5,
      "marginRight": -5,
    }
  }
>
  <div
    className="spt-col-4"
    style={
      Object {
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    }
  >
    Left Col
  </div>
  <div
    className="spt-col-20"
    style={
      Object {
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    }
  >
    Right Col
  </div>
</div>
`;
